lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -552,13 +552,16 @@ def success_from(action, response)
         response['response_summary'] == 'Approved' || response['approved'] == true || !response.key?('response_summary') && response.key?('action_id')
       end
 
-      def message_from(succeeded, response)
+      def message_from(succeeded, response, options)
         if succeeded
           'Succeeded'
         elsif response['error_type']
           response['error_type'] + ': ' + response['error_codes'].first
+        elsif options[:threeds_response_message]
+          response_summary = response['response_summary'] || response.dig('actions', 0, 'response_summary')
+          response_summary || response['status'] || response['message'] || 'Unable to read error message'
         else
-          response['response_summary'] || response['response_code'] || response['status'] || response['message'] || 'Unable to read error message'
+          response['response_summary'] || response['status'] || response['message'] || 'Unable to read error message'
```

Comment:
Why drop `response_code` from possible message options?

Reply:
Great catch, didn't realize I had removed it. Will update!